std/file.d

```diff
@@ -946,13 +947,96 @@ if (isConvertibleToString!RF || isConvertibleToString!RT)
     assert(t2.readText == "2");
 }
 
-private void renameImpl(const(char)[] f, const(char)[] t, const(FSChar)* fromz, const(FSChar)* toz) @trusted
+@safe unittest
+{
+    import std.utf : byWchar;
+
+    auto t1 = deleteme, t2 = deleteme~"2";
+    scope(exit) foreach (t; [t1, t2]) if (t.exists) t.remove();
+
+    write(t1, "1");
+    rename(t1, t2);
+    assert(readText(t2) == "1");
+
+    write(t1, "2");
+    rename(t1, t2.byWchar);
+    assert(readText(t2) == "2");
+}
+
+
+/***************************************************
+Move file `from` _to `to`.
+If the target file exists, it is overwritten. Unlike `rename` this function supports
+moving files accross drives/filesystems.
+Params:
+   from = string or range of characters representing the existing file name
+   to = string or range of characters representing the target file name
+Throws: `FileException` on error.
+*/
+void moveFile(RF, RT)(RF from, RT to)
+if ((isInputRange!RF && !isInfinite!RF && isSomeChar!(ElementEncodingType!RF) || isSomeString!RF)
+    && !isConvertibleToString!RF &&
+    (isInputRange!RT && !isInfinite!RT && isSomeChar!(ElementEncodingType!RT) || isSomeString!RT)
+    && !isConvertibleToString!RT)
```

Comment:
We really need a good `isStringLike` trait :/
See e.g. https://github.com/dlang/phobos/pull/5137